askama_shared/src/input.rs

```diff
@@ -28,23 +28,25 @@ impl TemplateInput<'_> {
         ast: &'n syn::DeriveInput,
         config: &'n Config<'_>,
     ) -> Result<TemplateInput<'n>, CompileError> {
-        // Check that an attribute called `template()` exists and that it is
+        // Check that an attribute called `template()` exists once and that it is
         // the proper type (list).
-        let meta = ast
-            .attrs
-            .iter()
-            .find_map(|attr| {
-                attr.path.is_ident("template").then(|| {
-                    attr.parse_meta()
-                        .map_err(|e| format!("unable to parse attribute: {}", e).into())
-                })
-            })
-            .unwrap_or(Err(CompileError::Static("no attribute 'template' found")))?;
-
-        let meta_list = match meta {
-            syn::Meta::List(inner) => inner,
-            _ => return Err("attribute 'template' has incorrect type".into()),
-        };
+        let mut template_args = None;
+        for attr in &ast.attrs {
+            if attr.path.is_ident("template") {
+                if template_args.is_some() {
+                    return Err(CompileError::Static("duplicated 'template' attribute"));
+                }
+                let template = attr.parse_meta().map_err(|e| {
+                    CompileError::String(format!("unable to parse attribute: {}", e))
+                })?;
+                match template {
+                    syn::Meta::List(inner) => template_args = Some(inner),
+                    _ => return Err("attribute 'template' has incorrect type".into()),
```

Comment:
The rests of the code uses `CompileError::Static` explicitly. For consistency, should we do that here as well?

Reply:
Yes, it does look more consistent this way. The only bad think is that the line is now longer than 100 characters, and "cargo fmt" even added a block around the return statement, turning this one line into five. :)

Reply:
Ah... Damn :)